### ISSUE TITLE
white screen on installing on typo3v12.4.11

### ISSUE THREAD

**Issue by rblezer** [FriendsOfTYPO3/extension_builder#734]
white screen on installing extension builder onto website via composer.

TYPO3 Version 12.4.11
Extension builder v12.0.0-beta.1

![image](https://github.com/FriendsOfTYPO3/extension_builder/assets/6106699/da00664a-a807-4326-aa46-03d913e6616a)

**Comment by DavidBruchmann** [comment]
Error messages from the browser console would certainly be interesting.

**Comment by rblezer** [comment]
Hi David,

Thanks for the quick response,
this is  the console output:

This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. main<http://landingpage.local/typo3/main?token=5ac9b0a3d18e6c9347e925bbe07644ddb4c48557>
Successfully compiled asm.js code (total compilation time 2ms) es-module-shims.js<http://landingpage.local/_assets/1ee1d3e909b58d32e30dcea666dd3224/JavaScript/Contrib/es-module-shims.js>
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. extensionBuilder<http://landingpage.local/typo3/module/tools/extensionBuilder?token=fbc8e770af8fd0efa1529e8d6a258b52a070ab1c>
Successfully compiled asm.js code (total compilation time 3ms) es-module-shims.js<http://landingpage.local/_assets/1ee1d3e909b58d32e30dcea666dd3224/JavaScript/Contrib/es-module-shims.js>
Content-Security-Policy: The page’s settings blocked the loading of a resource at eval (“script-src”). main.js:557<http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd>
Uncaught (in promise) EvalError: call to eval() blocked by CSP
    js http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:557
    __webpack_require__ http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:3229
    <anonymous> http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:3500
    <anonymous> http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:3503
    webpackUniversalModuleDefinition http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:15
    <anonymous> http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:18
main.js:557:1<http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/main.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd>
Hello from extensionbuilder.js extensionbuilder.js:1:9<http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/extensionbuilder.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd>
Uncaught (in promise) TypeError: slackButton is null
    <anonymous> http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/extensionbuilder.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd:118
extensionbuilder.js:118:1<http://landingpage.local/_assets/705f31da96f0870eac9db412ee92df1f/JavaScript/extensionbuilder.js?bust=cdfd7a8e1874e28dd032c511f51447eac22b8bcd>
[X]

________________________________

**Comment by PKuhlmay** [comment]
Thanks, I will have a look at it. Do you have some more information? PHP Version, Composer mode?, Production or Development Mode? Content Security Policy enabled or disabled?
The error looks like a JavaScript error but these information could help me with debugging the error as I don't get this one.

**Comment by rblezer** [comment]
It's a development site at a local VM, so it's a nonproduction site.
As I am battling for weeks to get a simple form to work, with hidden fields which contain data form tt_content, I thought I give content builder a try. The content elements with DCE are quite a Succes.
In Joomla it would be ready in hours.

anyway:

PHP version 8.3.3
Composer version 2.5.1

Content Security Policy

The following features flags are currently disabled and won't be shown in this module:

  *   security.frontend.enforceContentSecurityPolicy

no entries available


________________________________

**Comment by PKuhlmay** [comment]
This could be because of the content security policy. Could you try to disable it for the backend? In the Admin-Settings under Settings -> Feature Toggle.
I get a blank screen, when I enable the CSP, too. But the errors are a little bit different.

**Comment by rblezer** [comment]
I'll try that tomorrow, when I'm back in the office.Thanks allot.
-------- Oorspronkelijk bericht --------Van: Philipp Kuhlmay ***@***.***> Datum: 05-03-2024  16:55  (GMT+01:00) Aan: FriendsOfTYPO3/extension_builder ***@***.***> Cc: Richard Blezer ***@***.***>, Author ***@***.***> Onderwerp: Re: [FriendsOfTYPO3/extension_builder] white screen on installing on
  typo3v12.4.11 (Issue #734) 
This could be because of the content security policy. Could you try to disable it for the backend? In the Admin-Settings under Settings -> Feature Toggle.
I get a blank screen, when I enable the CSP, too. But the errors are a little bit different.

**Comment by rblezer** [comment]
Yes Succes,

when this setting is enabled, page turns blank.

Security: backend enforce content security policy

Iif on, HTTP Content-Security-Policy header will be applied for each HTTP backend request.
Default setting: Disabled





________________________________
Van: Philipp Kuhlmay ***@***.***>
Verzonden: dinsdag 5 maart 2024 16:55
Aan: FriendsOfTYPO3/extension_builder ***@***.***>
CC: Richard Blezer ***@***.***>; Author ***@***.***>
Onderwerp: Re: [FriendsOfTYPO3/extension_builder] white screen on installing on typo3v12.4.11 (Issue #734)


This could be because of the content security policy. Could you try to disable it for the backend? In the Admin-Settings under Settings -> Feature Toggle.
I get a blank screen, when I enable the CSP, too. But the errors are a little bit different.

**Comment by PKuhlmay** [comment]
Thanks for testing. The extension builder should also work, when content security policy is enabled. I think the problem occurs, when I use the wrong command for building all the JS assets with Webpack.

When I compile all the assets in Development mode, it fails, when I compile them in Production mode, it works.

This issue could be duplicated from #669

**Comment by DavidBruchmann** [comment]
I can confirm the bug and the positive impact of the feature toggle.
without backtrace my error was this:
> Uncaught (in promise) EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'nonce-aLrV9gzsze8oMu90WBgdNn9X7EjAhTKaGOSux1VckgOlUB7ZkgLP_A' 'report-sample'".

**Comment by PKuhlmay** [comment]
Closing this issue as it is no longer relevant.

The white screen was caused by Webpack's development mode bundling strategy, which conflicts with TYPO3's Content Security Policy. This was a known limitation of the React/Webpack-based GUI that was being developed for v12.

That React-based GUI has since been **discarded** and the domain modeling editor has been completely **rebuilt using Lit JS and Vite** in the current `master` branch. Vite's production output does not use the problematic bundling approach, so this CSP conflict no longer exists.

If you encounter any issues with the current version, please open a new ticket.
